src/tests/src/solver/modelParser/testModelParser.cpp

```diff
@@ -0,0 +1,264 @@
+
+#include <boost/test/unit_test.hpp>
+
+#include "antares/solver/modelParser/parser.h"
+
+// Test empty library
+BOOST_AUTO_TEST_CASE(test_empty_library)
+{
+    Antares::Solver::ModelParser::Parser parser;
+    auto library = R"(
+        library:
+            id: ""
+            description: ""
+            port-types: []
+            models: []
+    )";
+    Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
+    BOOST_CHECK(libraryObj.id.empty());
+    BOOST_CHECK(libraryObj.description.empty());
+    BOOST_CHECK(libraryObj.port_types.empty());
+    BOOST_CHECK(libraryObj.models.empty());
+}
+
+// Test library with id and description
+BOOST_AUTO_TEST_CASE(test_library_id_description)
+{
+    Antares::Solver::ModelParser::Parser parser;
+    auto library = R"(
+        library:
+            id: "test_id"
+            description: "test_description"
+            port-types: []
+            models: []
+        )";
+    Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
+    BOOST_CHECK(libraryObj.id == "test_id");
+    BOOST_CHECK(libraryObj.description == "test_description");
+    BOOST_CHECK(libraryObj.port_types.empty());
+    BOOST_CHECK(libraryObj.models.empty());
+}
+
+// Test library with port types
+BOOST_AUTO_TEST_CASE(test_library_port_types)
+{
+    Antares::Solver::ModelParser::Parser parser;
+    auto library = R"(
+        library:
+            id: "lib_id"
+            description: "lib_description"
+            port-types:
+                - id: "porttype_id"
+                  description: "porttype_description"
+                  fields:
+                      - name: "port_name"
+            models: []
+        )";
+    Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
+    BOOST_CHECK(libraryObj.port_types.size() == 1);
+    BOOST_CHECK(libraryObj.port_types[0].id == "porttype_id");
+    BOOST_CHECK(libraryObj.port_types[0].description == "porttype_description");
+    BOOST_CHECK(libraryObj.port_types[0].fields.size() == 1);
+    BOOST_CHECK(libraryObj.port_types[0].fields[0] == "port_name");
+}
+
+// Test library with models
+BOOST_AUTO_TEST_CASE(test_library_models)
+{
+    Antares::Solver::ModelParser::Parser parser;
+    auto library = R"(
+        library:
+            id: "lib_id"
+            description: "lib_description"
+            port-types: []
+            models:
+                - id: "model_id"
+                  description: "model_description"
+                  parameters: []
+                  variables: []
+                  ports: []
+                  port-field-definitions: []
+                  constraints: []
+                  objective: "objective"
+        )";
+    Antares::Solver::ModelParser::Library libraryObj = parser.parse(library);
+    BOOST_CHECK(libraryObj.models.size() == 1);
+    BOOST_CHECK(libraryObj.models[0].id == "model_id");
+    BOOST_CHECK(libraryObj.models[0].description == "model_description");
```

Comment:
Use `BOOST_REQUIRE` to exit in case of invalid size (avoid subsequent segfault)

```suggestion
    BOOST_REQUIRE_EQUAL(libraryObj.models.size(), 1);
    BOOST_CHECK_EQUAL(libraryObj.models[0].id, "model_id");
    BOOST_CHECK_EQUAL(libraryObj.models[0].description, "model_description");
```